File: samples/csharp/end-to-end-apps/MulticlassClassification-GitHubLabeler/GitHubLabeler/GitHubLabelerConsoleApp/GitHubLabelerTextLoaderFactory.cs
using Microsoft.ML;
using Microsoft.ML.Core.Data;
using Microsoft.ML.Runtime.Data;
using Microsoft.ML.Transforms;
using System;
using System.Collections.Generic;
using System.Text;

namespace GitHubLabeler
{
    public static class GitHubLabelerTextLoaderFactory
    {
        public static TextLoader CreateTextLoader(MLContext mlContext)
        {
            TextLoader textLoader = mlContext.Data.TextReader(new TextLoader.Arguments()
                                                    {
                                                        Separator = "tab",
                                                        HasHeader = true,
                                                        Column = new[]
                                                                    {
                                                                        new TextLoader.Column("ID", DataKind.Text, 0),
                                                                        new TextLoader.Column("Area", DataKind.Text, 1),
                                                                        new TextLoader.Column("Title", DataKind.Text, 2),
                                                                        new TextLoader.Column("Description", DataKind.Text, 3),
                                                                    }
                                                    });

            return textLoader;
        }
    }

}

File: samples/fsharp/end-to-end-apps/MulticlassClassification-GitHubLabeler/GitHubLabeler/GitHubLabelerConsoleApp/GitHubLabelerTextLoaderFactory.fs
module GitHubLabelerTextLoaderFactory

open System
open Microsoft.ML
open Microsoft.ML.Runtime.Data


let createTextLoader (mlContext : MLContext) =
    mlContext.Data.TextReader(
        TextLoader.Arguments(
            Separator = "tab",
            HasHeader = true,
            Column = 
                [|

                    TextLoader.Column("ID", Nullable DataKind.Text, 0)
                    TextLoader.Column("Area", Nullable DataKind.Text, 1)
                    TextLoader.Column("Title", Nullable DataKind.Text, 2)
                    TextLoader.Column("Description", Nullable DataKind.Text, 3)
                |]
        )
    )

let read datasetPath (textLoader : TextLoader) =
    textLoader.Read([| datasetPath |])